source/PlayerInfo.cpp

```diff
@@ -1773,35 +1773,41 @@ bool PlayerInfo::TakeOff(UI *ui, const bool distributeCargo)
 	for(const Mission *mission : missionsToRemove)
 		RemoveMission(Mission::ABORT, *mission, ui);
 
-	// Any ordinary cargo left behind can be sold.
+	// Any ordinary cargo/outfits left behind can be either sold, stored or dumped.
 	int64_t income = 0;
 	int day = date.DaysSinceEpoch();
-	int64_t sold = cargo.Used();
+	int64_t leftOver = cargo.Used();
 	int64_t totalBasis = 0;
-	if(sold)
+	if(leftOver)
 	{
-		for(const auto &commodity : cargo.Commodities())
+		bool canSell = planet->CanUseServices() && system->HasTrade()
+			&& planet->GetPort().HasService(Port::ServicesType::Trading);
+
+		if(canSell)
 		{
-			if(!commodity.second)
-				continue;
+			for(const auto &commodity : cargo.Commodities())
+			{
+				if(!commodity.second)
+					continue;
 
-			// Figure out how much income you get for selling this cargo.
-			int64_t value = commodity.second * static_cast<int64_t>(system->Trade(commodity.first));
-			income += value;
+				// Figure out how much income you get for selling this cargo.
+				int64_t value = commodity.second * static_cast<int64_t>(system->Trade(commodity.first));
+				income += value;
 
-			int original = originalTotals[commodity.first];
-			auto it = costBasis.find(commodity.first);
-			if(!original || it == costBasis.end() || !it->second)
-				continue;
+				int original = originalTotals[commodity.first];
+				auto it = costBasis.find(commodity.first);
+				if(!original || it == costBasis.end() || !it->second)
+					continue;
 
-			// Now, figure out how much of that income is profit by calculating
-			// the cost basis for this cargo (which is just the total cost basis
-			// multiplied by the percent of the cargo you are selling).
-			int64_t basis = it->second * commodity.second / original;
-			it->second -= basis;
-			totalBasis += basis;
+				// Now, figure out how much of that income is profit by calculating
+				// the cost basis for this cargo (which is just the total cost basis
+				// multiplied by the percent of the cargo you are selling).
+				int64_t basis = it->second * commodity.second / original;
+				it->second -= basis;
+				totalBasis += basis;
+			}
 		}
-		if(!planet->HasOutfitter())
+		if(!planet->HasOutfitter() && canSell)
```

Comment:
You can also change this one to
```
if(planet->HasOutfitter())
{
	if(canUseServices)
		{code for storing}
}
else if(canSell)
	{code for selling}
```
to avoid doing the outfitter check twice.

And, there's a `canUseServices` var defined at the beginning of this function, so replace all `planet->CanUseServices()` instances you have here with the var.